tests/integration/targets/ndo_l3out_interface_group_policy/tasks/main.yml

```diff
@@ -0,0 +1,604 @@
+        <<: *mso_info
+        template: ansible_tenant_template
+        name: ansible_l3out_interface_routing_policy
+        bfd_settings:
+          state: enabled
+        bfd_multi_hop_settings:
+          state: enabled
+        state: present
+      register: interface_routing_policy
+
+    - name: Create a new Custom QoS Policy object
+      cisco.mso.ndo_tenant_custom_qos_policy:
+        <<: *mso_info
+        template: ansible_tenant_template
+        name: ansible_custom_qos_policy
+        dscp_mappings:
+          - dscp_from: af11
+            dscp_to: af12
+            dscp_target: af11
+            target_cos: background
+            qos_priority: level1
+        cos_mappings:
+          - dot1p_from: background
+            dot1p_to: best_effort
+            target: af11
+            target_cos: background
+            qos_priority: level1
+        state: present
+      register: custom_qos_policy
+
+    # L3Out Template Setup
+    - name: Create a new l3out template
+      cisco.mso.ndo_template:
+        <<: *ndo_l3out_template_absent
+        state: present
+      register: create_new_l3out_policy_template
+
+    - name: Create l3out_1 object with routing protocols
+      cisco.mso.ndo_l3out_template:
+        <<: *mso_info
+        l3out_template: '{{ mso_l3out_template | default("ansible_test") }}'
+        name: l3out_1
+        vrf:
+          name: VRF1
+          schema: '{{ mso_schema | default("ansible_test") }}'
+          template: ansible_template
+        bgp:
+          state: enabled
+        ospf:
+          state: enabled
+          area_id: 1
+          area_type: regular
+        state: present
+      register: create_l3out_1_object
+
+    - name: Create L3Out interface group policy object - check mode
+      cisco.mso.ndo_l3out_interface_group_policy: &cm_interface_group_policy_1_present
+        <<: *mso_info
+        template: '{{ mso_l3out_template | default("ansible_test") }}'
+        l3out: l3out_1
+        name: interface_group_policy_1
+        state: present
+      check_mode: true
+      register: cm_interface_group_policy_1_present
+
+    - name: Assertion check for create L3Out interface group policy object with default values - check mode
+      ansible.builtin.assert:
+        that:
+          - cm_interface_group_policy_1_present is changed
+          - cm_interface_group_policy_1_present.current.name == "interface_group_policy_1"
+          - cm_interface_group_policy_1_present.previous == {}
+
+    - name: Create L3Out interface group policy object with default values - normal mode
+      cisco.mso.ndo_l3out_interface_group_policy: &nm_interface_group_policy_1_present
+        <<: *cm_interface_group_policy_1_present
+      register: nm_interface_group_policy_1_present
+
+    - name: Assertion check for create L3Out interface group policy object with default values - normal mode
+      ansible.builtin.assert:
+        that:
+          - nm_interface_group_policy_1_present is changed
```

Comment:
Should we add an assertion for `templateName` and `templateId` in all the test cases?